app/models/roster_entry.rb

```diff
@@ -10,19 +10,23 @@ class IdentifierCreationError < StandardError; end
 
   before_create :validate_identifiers_are_unique_to_roster
 
-  def self.to_csv
+  # rubocop:disable Metrics/MethodLength, Metrics/AbcSize
+  def self.to_csv(user_to_group_map = {})
     CSV.generate(headers: true, col_sep: ",", force_quotes: true) do |csv|
-      csv << %i[identifier github_username github_id name]
+      csv << %i[identifier github_username github_id name group_name]
 
       all.sort_by(&:identifier).each do |entry|
         github_user = entry.user.try(:github_user)
-        github_id = github_user.try(:id) || ""
         login = github_user.try(:login) || ""
+        github_id = github_user.try(:id) || ""
         name = github_user.try(:name) || ""
-        csv << [entry.identifier, login, github_id, name]
+        group_name = user_to_group_map.empty? ? "" : user_to_group_map[entry.user_id]
+
+        csv << [entry.identifier, login, github_id, name, group_name]
```

Comment:
Similarly to above we, could remove the group name column from the row by doing the following:
```ruby
row = [entry.identifier, login, github_id, name]
row << group_name unless group_name.empty?
csv << row
```